types/collection.hh

```diff
@@ -123,9 +132,10 @@ collection_type_impl::pack(BytesViewIterator start, BytesViewIterator finish, in
     return out;
 }
 
-template <typename BytesViewIterator>
+template <typename Iterator>
+requires requires (Iterator it) { {*it} -> std::convertible_to<bytes_view>; }
```

Comment:
Is it not a BytesViewIterator now? Do not understand the name change.

Reply:
It is an iterator over either `bytes_view` or `bytes`, so the name wasn't great in the first place. The constraint expresses the intent better than the name.

Reply:
Ok. We can introduce a named concept for this thing, but that can be done later.